src/metldata/builtin_transformations/copy_content/data_transform.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Copy content properties between resources described by a relation path."""
+
+from schemapack.spec.datapack import DataPack
+
+from metldata.builtin_transformations.add_content_properties.path import (
+    resolve_data_object_path,
+)
+from metldata.builtin_transformations.common.custom_types import (
+    MutableClassResources,
+    MutableResource,
+    ResourceId,
+)
+from metldata.builtin_transformations.common.data_transform import get_class_resources
+from metldata.builtin_transformations.common.resolve_path import resolve_path
+from metldata.builtin_transformations.common.utils import data_to_dict
+from metldata.builtin_transformations.copy_content.instruction import (
+    CopyContentInstruction,
+)
+from metldata.transform.exceptions import EvitableTransformationError
+
+
+class ContentCopier:
+    """Helper for content copying."""
+
+    def __init__(
+        self,
+        *,
+        data: DataPack,
+        instructions_by_class: dict[str, list[CopyContentInstruction]],
+    ):
+        self.instructions_by_class = instructions_by_class
+        self.data = data
+        self.modified_data = data_to_dict(self.data)
+
+    def process_instructions(self):
+        """Process high-level checks and delegate instruction processing."""
+        for class_name, instructions in self.instructions_by_class.items():
+            # Check resources exist for class to be modified
+            target_resources = get_class_resources(
+                data=self.modified_data, class_name=class_name
+            )
+            if not target_resources:
+                raise EvitableTransformationError()
+
+            for instruction in instructions:
+                self.process_instruction(
+                    target_resources=target_resources, instruction=instruction
+                )
+
+        return DataPack.model_validate(self.modified_data)
+
+    def process_instruction(
```

Comment:
I think the function name is confusing. It can be replaced with a name that is more representative of the function's responsibility on the overall workflow.